packages/components/layout/src/badge.tsx

```diff
@@ -21,18 +23,34 @@ export interface BadgeProps
 export const Badge = forwardRef<BadgeProps, "span">(function Badge(props, ref) {
   const styles = useStyleConfig("Badge", props)
   const { className, ...rest } = omitThemingProps(props)
+  const badgeStyle = useMemo((): ChakraProps => {
+    if (props.noOfLines) {
+      return {
+        __css: {
+          verticalAlign: "middle",
+          ...styles,
+        },
+        sx: {
+          display: props.display || "-webkit-inline-box",
```

Comment:
The `-webkit-line-clamp` CSS property only works in combination with the display property set to `-webkit-box` or `-webkit-inline-box`. In this case, it should use `-webkit-inline-box` instead of `-webkit-box`.

https://developer.mozilla.org/en-US/docs/Web/CSS/-webkit-line-clamp